packages/mui-joy/src/Button/Button.tsx

```diff
@@ -1,21 +1,258 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { unstable_capitalize as capitalize, unstable_useForkRef as useForkRef } from '@mui/utils';
+import { useButton } from '@mui/base/ButtonUnstyled';
+import composeClasses from '@mui/base/composeClasses';
+import { styled, useThemeProps } from '../styles';
+import { rootShouldForwardProp } from '../styles/styled';
+import { ExtendButton, ButtonTypeMap, ButtonProps } from './ButtonProps';
+import buttonClasses, { getButtonUtilityClass } from './buttonClasses';
 
-export type ButtonProps = JSX.IntrinsicElements['button'];
+const useUtilityClasses = (ownerState: ButtonProps & { focusVisible: boolean }) => {
+  const {
+    classes,
+    color,
+    disabled,
+    focusVisible,
+    focusVisibleClassName,
+    fullWidth,
+    size,
+    variant,
+  } = ownerState;
 
-const Button = (props: ButtonProps) => {
-  return <button type="button" {...props} />;
+  const slots = {
+    root: [
+      'root',
+      disabled && 'disabled',
+      focusVisible && 'focusVisible',
+      fullWidth && 'fullWidth',
+      `variant${capitalize(variant!)}`,
+      `color${capitalize(color!)}`,
+      `size${capitalize(size!)}`,
+    ],
+  };
+
+  const composedClasses = composeClasses(slots, getButtonUtilityClass, classes);
+
+  if (focusVisible && focusVisibleClassName) {
+    composedClasses.root += ` ${focusVisibleClassName}`;
+  }
+
+  return composedClasses;
 };
 
+const ButtonRoot = styled('button', {
+  name: 'JoyButton',
+  slot: 'Root',
+  overridesResolver: (props, styles) => {
```

Comment:
Are we going to use CSS variables for overrides?

Reply:
If you mean specific API for CSS variables, I don't think we need it at the beginning. I will just go with what we have and treat CSS variables as CSS properties.